### ISSUE TITLE
TwilightForest error spam & player disconnect/crash.

### ISSUE THREAD

**Issue by Zilacon** [PFCraft/Mohist#25]
This has been an issue since i tried mohist on 0.0.7. This is being tested on the Tekxit modpack.
When launching theres so many twilightforest errors changing from yellow to red that its like a epileptic seizure on the console. These errors get spammed all throughout the start up from mods, worlds, and plugins. And then when players login and play on the world.

If this decides to pop up (happens in any overworld, have not tested nether or end) it will make that region of the world a crash zone until the region is deleted. That was literally the only way we could rejoin that section of the world is to wipe the world region files. Restarting/rejoining/wiping player data did nothing.

Heres a screenshot of the error in question, followed by the entire console log.
![Screenshot_155](https://user-images.githubusercontent.com/3002248/58372212-37bd6580-7ecf-11e9-9b63-15c063a81e4b.png)
[latest.log](https://github.com/PFCraft/Mohist/files/3219523/latest.log)

**Comment by Zilacon** [comment]
Any update on this?

**Comment by braddevans** [comment]
```java
[01:45:21 ERROR]: Cannot Hotload Dim: 5
[01:45:21 ERROR]: FMLIndexedMessageCodec exception caught
java.lang.NullPointerException: null
	at net.minecraftforge.fml.common.network.FMLOutboundHandler$OutboundTarget$9.selectNetworks(FMLOutboundHandler.java:304) ~[FMLOutboundHandler$OutboundTarget$9.class:0.0.9a-6e0f02a]
	at net.minecraftforge.fml.common.network.FMLOutboundHandler.write(FMLOutboundHandler.java:378) ~[FMLOutboundHandler.class:0.0.9a-6e0f02a]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:738) ~[AbstractChannelHandlerContext.class:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:730) ~[AbstractChannelHandlerContext.class:?]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:816) ~[AbstractChannelHandlerContext.class:?]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:723) ~[AbstractChannelHandlerContext.class:?]
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:111) ~[MessageToMessageEncoder.class:?]
	at io.netty.handler.codec.MessageToMessageCodec.write(MessageToMessageCodec.java:116) ~[MessageToMessageCodec.class:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:738) ~[AbstractChannelHandlerContext.class:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:801) ~[AbstractChannelHandlerContext.class:?]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:814) ~[AbstractChannelHandlerContext.class:?]
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:794) ~[AbstractChannelHandlerContext.class:?]
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:831) ~[AbstractChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1032) ~[DefaultChannelPipeline.class:?]
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:296) ~[AbstractChannel.class:?]
	at net.minecraftforge.fml.common.network.simpleimpl.SimpleNetworkWrapper.sendToAllTracking(SimpleNetworkWrapper.java:295) [SimpleNetworkWrapper.class:0.0.9a-6e0f02a]
	at twilightforest.capabilities.shield.ShieldCapabilityHandler.sendUpdatePacket(ShieldCapabilityHandler.java:93) [ShieldCapabilityHandler.class:?]
	at twilightforest.capabilities.shield.ShieldCapabilityHandler.setShields(ShieldCapabilityHandler.java:74) [ShieldCapabilityHandler.class:?]
	at twilightforest.capabilities.shield.ShieldCapabilityStorage.readNBT(ShieldCapabilityStorage.java:25) [ShieldCapabilityStorage.class:?]
	at twilightforest.capabilities.shield.ShieldCapabilityStorage.readNBT(ShieldCapabilityStorage.java:10) [ShieldCapabilityStorage.class:?]
	at twilightforest.capabilities.CapabilityList$1.deserializeNBT(CapabilityList.java:62) [CapabilityList$1.class:?]
	at twilightforest.capabilities.CapabilityList$1.deserializeNBT(CapabilityList.java:37) [CapabilityList$1.class:?]
	at net.minecraftforge.common.capabilities.CapabilityDispatcher.deserializeNBT(CapabilityDispatcher.java:135) [CapabilityDispatcher.class:0.0.9a-6e0f02a]
	at net.minecraft.entity.Entity.func_70020_e(Entity.java:1981) [vg.class:?]
	at net.minecraft.entity.EntityList.func_75615_a(EntityList.java:222) [vi.class:?]
	at net.minecraft.world.chunk.storage.AnvilChunkLoader.func_186053_a(AnvilChunkLoader.java:599) [aye.class:?]
	at net.minecraft.world.chunk.storage.AnvilChunkLoader.func_186050_a(AnvilChunkLoader.java:526) [aye.class:?]
	at net.minecraftforge.common.ForgeChunkManager.loadChunkEntities(ForgeChunkManager.java:1028) [ForgeChunkManager.class:0.0.9a-6e0f02a]
	at net.minecraftforge.common.ForgeChunkManager.fetchDormantChunk(ForgeChunkManager.java:1017) [ForgeChunkManager.class:0.0.9a-6e0f02a]
	at net.minecraft.world.gen.ChunkProviderServer.loadChunk(ChunkProviderServer.java:107) [on.class:?]
	at net.minecraft.server.management.PlayerChunkMapEntry.<init>(PlayerChunkMapEntry.java:46) [ot.class:?]
	at net.minecraft.server.management.PlayerChunkMap.func_187302_c(PlayerChunkMap.java:244) [ou.class:?]
	at net.minecraft.server.management.PlayerChunkMap.func_72683_a(PlayerChunkMap.java:302) [ou.class:?]
	at net.minecraft.server.management.PlayerList.recreatePlayerEntity(PlayerList.java:827) [pl.class:?]
	at org.bukkit.craftbukkit.v1_12_R1.entity.CraftPlayer.teleport(CraftPlayer.java:626) [CraftPlayer.class:0.0.9a-6e0f02a]
	at com.earth2me.essentials.paperlib.features.asyncteleport.AsyncTeleportSync.teleportAsync(AsyncTeleportSync.java:13) [EssentialsX-2.16.1.154.jar:?]
	at com.earth2me.essentials.paperlib.environments.Environment.teleport(Environment.java:79) [EssentialsX-2.16.1.154.jar:?]
	at com.earth2me.essentials.paperlib.PaperLib.teleportAsync(PaperLib.java:86) [EssentialsX-2.16.1.154.jar:?]
	at com.earth2me.essentials.Teleport.now(Teleport.java:143) [EssentialsX-2.16.1.154.jar:?]
	at com.earth2me.essentials.Teleport.teleport(Teleport.java:202) [EssentialsX-2.16.1.154.jar:?]
	at com.earth2me.essentials.Teleport.warp(Teleport.java:299) [EssentialsX-2.16.1.154.jar:?]
	at com.earth2me.essentials.commands.Commandwarp.warpUser(Commandwarp.java:103) [EssentialsX-2.16.1.154.jar:?]
	at com.earth2me.essentials.commands.Commandwarp.run(Commandwarp.java:47) [EssentialsX-2.16.1.154.jar:?]
	at com.earth2me.essentials.commands.EssentialsCommand.run(EssentialsCommand.java:161) [EssentialsX-2.16.1.154.jar:?]
	at com.earth2me.essentials.Essentials.onCommandEssentials(Essentials.java:545) [EssentialsX-2.16.1.154.jar:?]
	at com.earth2me.essentials.Essentials.onCommand(Essentials.java:458) [EssentialsX-2.16.1.154.jar:?]
	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44) [PluginCommand.class:0.0.9a-6e0f02a]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:144) [SimpleCommandMap.class:0.0.9a-6e0f02a]
	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchCommand(CraftServer.java:688) [CraftServer.class:0.0.9a-6e0f02a]
	at net.minecraft.network.NetHandlerPlayServer.func_147361_d(NetHandlerPlayServer.java:1570) [pa.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147354_a(NetHandlerPlayServer.java:1417) [pa.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:37) [la.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:9) [la.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) [hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_212]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_212]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:831) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:444) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:774) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:632) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:813) [?:1.8.0_212]
```

**Comment by Zilacon** [comment]
This error IS PRESENT on normal java 8 and on latest 1.2 version.

**Comment by Zilacon** [comment]
This seems to be resolved since #346
